src/main.cpp

```diff
@@ -4515,12 +4574,21 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         }
 
         CBlockIndex *pindexLast = NULL;
+        std::vector<CBlockIndex *> vToFetch;
+        bool fCanDirectFetch = CanDirectFetch(chainparams.GetConsensus());
```

Comment:
This block of code already has a guard so that we don't process headers received while importing/reindexing.